napari/plugins/manager.py

```diff
@@ -26,6 +26,17 @@
 
 pluggy.manager._HookCaller = _HookCaller
 
+# the main plugin manager instance for the `napari` plugin namespace.
+_plugin_manager = None
+
+
+# TODO: ask Juan if there's a better pattern than this for getting a lazy singleton
```

Comment:
:joy: I literally hunted down this specific commit going, "why did he do this? There *must* be a better way." :joy: I honestly hate this pattern. But I can't find the motivation for this... Is it to avoid instantiating the plugin manager until... Until what? Why don't we want to instantiate the plugin manager?

Reply:
This page has a lot of options for singleton objects, and I like all of them better than a global `get_...()` function:

https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html

Reply:
Probably my favourite is the class variable approach:

```python
# Singleton/ClassVariableSingleton.py
class SingleTone(object):
    __instance = None
    def __new__(cls, val):
        if SingleTone.__instance is None:
            SingleTone.__instance = object.__new__(cls)
        SingleTone.__instance.val = val
        return SingleTone.__instance
```

Reply:
Well actually what we’re really trying to delay is the plugin discovery... but that’s definitely also worth discussing.  We were reasoning that since it imports all the plugin modules and might add to startup, perhaps we want to wait until someone tried to actually use a plugin.  But what do you think?

Reply:
Could that be delayed by having a single object like we have now, and then having `.hooks` (or wherever the plugin discovery *needs* to have happened) be a property that triggers the plugin discovery?

Reply:
yep! was actually thinking the same thing